### DISCUSSION TITLE
Model Collection Not Displaying On Blades

### DISCUSSION THREAD

**Discussion by dev-techguy** [livewire/livewire#4225] [7. Bugs (archived)]
Octane Version: 1.0.19
Laravel Version: 8.71.0
PHP Version: 8.0.12
Server & Version: Swoole
Database Driver & Version: postgres 12
No data is being displayed on the blade.

Return a model collection on a blade and try to loop through via a table.

**Comment by joshhanley** [comment]
@dev-techguy please share your component class and blade view as well as any other relevant code.

Thanks!

**Reply by dev-techguy** [reply]
KIndly check the Defer Loading feature.

**Comment by dev-techguy** [comment]
`<?php

namespace App\Http\Livewire\User\Manage;

use Jantinnerezo\LivewireAlert\LivewireAlert;
use LaravelMultipleGuards\Traits\FindGuard;
use Livewire\Component;
use Livewire\WithPagination;

class Users extends Component
{
    use WithPagination, FindGuard, LivewireAlert;

    public $readyToLoad = false;
    public $search;
    public $team_user_id;

    protected $queryString = ['search'];

    protected $paginationTheme = 'bootstrap';

    protected $listeners = [
        'confirmed',
        'cancelled'
    ];

    public function updatingSearch()
    {
        $this->resetPage();
    }

    public function loadUsers()
    {
        $this->readyToLoad = true;
    }

    public function deleteUser(string $id)
    {
        $this->team_user_id = $id;
        $this->confirm('Are you sure you want to proceed?', [
            'toast' => false,
            'position' => 'center',
            'showConfirmButton' => true,
            'confirmButtonText' => 'Yes, I am sure!',
            'cancelButtonText' => 'No, cancel it!',
            'onConfirmed' => 'confirmed',
            'onDismissed' => 'cancelled'
        ]);
    }

    public function confirmed()
    {
        $this->findGuardType()->user()->load('team.team_users')
            ->team
            ->team_users()
            ->findOrFail($this->team_user_id)
            ->forceDelete();

        $this->alert('success', 'You have removed user completely from the system.');
    }

    public function copyToClipBoard()
    {
        $this->dispatchBrowserEvent('copy_url');
    }


    public function cancelled()
    {
        $this->alert('error', 'You have canceled.');
    }


    public function render()
    {
        return view('livewire.user.manage.users', [
            'users' => $this->readyToLoad
                ? $this->findGuardType()->user()->load('team.team_users')
                    ->team
                    ->team_users()
                    ->latest('updated_at')
                    ->where(function ($query) {
                        $query->orWhere('name', 'ilike', '%' . $this->search . '%')
                            ->orWhere('email', 'ilike', '%' . $this->search . '%')
                            ->orWhere('phone_number', 'ilike', '%' . $this->search . '%');
                    })
                    ->paginate(env('MIN_PAGINATE'))
                : [],
            'team' => $this->findGuardType()->user()->load('team')->team
        ])->extends('layouts.user');
    }
}
`

'<div>
    <div class="main-content">
        <div class="breadcrumb">
            <ul>
                <li><a href="#">User Management</a></li>
                <li>{{ date('F d, Y', strtotime(now())) }}</li>
            </ul>
        </div>
        <div class="separator-breadcrumb border-top"></div>
        @livewire('user.inc.alert-notification')

        
        <div class="row mb-4">
            <div class="col-md-12 mb-4">
                <div class="card text-left">
                    @include('inc.alert')
                    <center>
                        <div class="loader-bubble loader-bubble-primary" wire:loading></div>
                    </center>
                    <div wire:init="loadUsers">
                        <div class="card-body">
                            <div wire:poll.{{ env('LOW_POLL') }}>
                                <div class="row">
                                    <div class="col-md-10">
                                        <div class="form-group">
                                            <input wire:model="search" type="search" class="form-control"
                                                   title="Search using unique keys depending on the data display"
                                                   placeholder="Search...">
                                        </div>
                                    </div>
                                    <div class="col-md-1">
                                        <div class="input-group mb-3">
                                            <input class="form-control" id="account_link" type="text" readonly
                                                   value="{{ route('team.login',['slug'=> $team->slug]) }}"/>
                                            <div class="input-group-append">
                                                <button class="btn btn-primary" wire:click="copyToClipBoard"
                                                        title="{{ route('team.login',['slug'=> $team->slug]) }}"
                                                        wire:offline.attr="disabled" wire:loading.attr="disabled">Copy
                                                    Link
                                                </button>
                                            </div>
                                        </div>
                                    </div>
                                    <div class="col-md-1">
                                        <a href="{{ route('add.user') }}" class="btn btn-primary float-right"><i
                                                class="nav-icon i-Add-User"></i><span
                                                class="item-name"> Add User</span></a>
                                    </div>
                                </div>
                                @if(count($users))
                                    <div class="table-responsive">
                                        <table class="display table table-striped table-bordered"
                                               id="zero_configuration_table"
                                               style="width:100%">
                                            <thead>
                                            <tr>
                                                <th>#</th>
                                                <th>Name</th>
                                                <th>Email</th>
                                                <th>Phone Number</th>
                                                <th>App Access</th>
                                                <th>Status</th>
                                                <th>Action</th>
                                                <th>Action</th>
                                                <th>Last Login</th>
                                                <th>Date/Time</th>
                                            </tr>
                                            </thead>
                                            <tbody>
                                            @php($count = 1)
                                            @foreach($users as $user)
                                                <tr>
                                                    <td>{{ $count++ }}</td>
                                                    <td>{{ $user->name }}</td>
                                                    <td>{{ $user->email }}</td>
                                                    <td>{{ $user->phone_number }}</td>
                                                    <td>{{ number_format(count($user->apps_access)) }}</td>
                                                    <td>
                                                        @if($user->is_suspended)
                                                            <span class="badge badge-pill badge-outline-danger p-2 m-1">suspended</span>
                                                        @else
                                                            <span
                                                                class="badge badge-pill badge-outline-success p-2 m-1">active</span>
                                                        @endif
                                                    </td>
                                                    <td><a href="{{ route('edit.user',['id'=> $user->id]) }}"
                                                           class="btn btn-sm btn-link"><span
                                                                class="nav-icon i-Pen-6"></span></a>
                                                    </td>
                                                    <td><a href="#" wire:click="deleteUser('{{ $user->id }}')"
                                                           class="btn btn-sm btn-danger">remove</a>
                                                    </td>
                                                    <td>
                                                        @if (!is_null($user->last_accessed_at))
                                                            {{ \App\Http\Controllers\SystemController::elapsedTime($user->last_accessed_at) }}
                                                        @else
                                                            Not Available
                                                        @endif
                                                    </td>
                                                    <td>{{ date('F d, Y h:i a', strtotime($user->created_at)) }}</td>
                                                </tr>
                                            @endforeach
                                            </tbody>
                                            <tfoot>
                                            <tr>
                                                <th>#</th>
                                                <th>Name</th>
                                                <th>Email</th>
                                                <th>Phone Number</th>
                                                <th>App Access</th>
                                                <th>Status</th>
                                                <th>Action</th>
                                                <th>Action</th>
                                                <th>Last Login</th>
                                                <th>Date/Time</th>
                                            </tr>
                                            </tfoot>
                                        </table>
                                    </div>
                                    <div class="row">
                                        <div class="col-md-12 text-right">
                                            {{ $users->links() }}
                                        </div>
                                    </div>
                                @else
                                    <hr>
                                    <center>
                                        @if(isset($search))
                                            <p class="text-center">No user with <b><u>{{ $search }}</u></b> details was
                                                found.</p>
                                            <a href="{{ route('user.manage.users') }}" class="btn btn-success"><i
                                                    class="nav-icon i-Arrow-Refresh"></i><span
                                                    class="item-name"> Refresh</span></a>
                                        @else
                                            <a href="{{ route('add.user') }}" class="btn btn-primary"><i
                                                    class="nav-icon i-Add-User"></i><span
                                                    class="item-name"> Add User</span></a>
                                        @endif
                                        <br>
                                        <img src="{{ asset('img/logo.png') }}" alt="" width="100">
                                    </center>
                                    <hr>
                                @endif
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        
        
    </div>
</div>
@push('scripts')
    <script>
        window.addEventListener('copy_url', event => {
            document.getElementById('account_link').select();
            document.execCommand('copy');
            Swal.fire(
                'Account Link',
                'Your account link has been copied to clipboard successfully.',
                'success'
            )
        })
    </script>
@endpush
'

**Reply by joshhanley** [reply]
@dev-techguy can you please fix the formatting of your code?

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Thanks!

**Reply by dev-techguy** [reply]
Thanks, @joshhanley i finally got the error.

**Reply by joshhanley** [reply]
Ok great! 🙂
